docs/src/pages/components/data-grid/editing/editing.md

```diff
@@ -5,7 +5,123 @@ components: DataGrid, XGrid
+
+### Stop Editing
+
+- <kbd class="key">Escape</kbd> to rollback changes.
+- <kbd class="key">Tab</kbd> to save and goes to next cell.
+- <kbd class="key">Enter</kbd> to save and goes to next cell below.
+- Click outside the cell
+
+### Control cell editability
+
+In addition to the `editable` flag on columns, control which cell is editable using `isCellEditable` prop.
+
+_In this demo, `Name` is not editable, and only the rows with an even `Age` value are editable.
+We applied a green background for a better visibility._
+
+{{"demo": "pages/components/data-grid/editing/IsCellEditableGrid.js", "bg": "inline"}}
+
+### Control editing
+
+The `editRowsModel` lets you control the editing.
+
+- implement the `onEditRowModelChange` to control the state of the `GridEditRowsModel` model.
+
+{{"demo": "pages/components/data-grid/editing/EditRowModelControlGrid.js", "bg": "inline", "defaultCodeOpen": false}}
+
+#### value Getter
+
+If the column definition process their data using a `valueGetter`.
+
+- set the underlying data by implementing `onEditCellChangeCommitted` event handler invoked when the change is submitted, and the grid is about to update.
+
+{{"demo": "pages/components/data-grid/editing/ValueGetterGrid.js", "bg": "inline"}}
+
+### Validation
+
+To validate cell values on input change.
+
+- set the event handler `onEditCellChange` invoked when a change is triggered by the edit input component.
+- set the event handler `onEditCellChangeCommitted` to validate when the change is committed.
+- or use the `GridEditRowsModel` mentioned in [Control Editing](#control-editing).
+
+{{"demo": "pages/components/data-grid/editing/ValidateRowModelControlGrid.js", "bg": "inline"}}
+
+#### using ApiRef
+
+If you purchased XGrid, it will be quicker to use `apiRef` to do that.
+{{"demo": "pages/components/data-grid/editing/ValidateCellApiRefGrid.js", "bg": "inline"}}
+
+### Server side
+
+Server side validation works just like client side [validation](#validation).
+
+- use the `GridEditRowsModel` mentioned in [Control Editing](#control-editing).
+- or set the event handler `onEditCellChange` for `keydown` validation
+- set the event handler `onEditCellChangeCommitted` to validate and update the server when the change is committed.
+
+**Note:** To prevent the default client side behavior, use `event.stopPropagation()`.
+
+_This demo shows how you can validate a username asynchronously and prevent the user from committing the value while validating._
+{{"demo": "pages/components/data-grid/editing/ValidateServerNameGrid.js", "bg": "inline", "defaultCodeOpen": false}}
+
+### Customization
+
+#### Component
+
+To customize the edit component of a column
+
+- set the `renderEditCell` function available in the column definition `GridColDef`.
+
+_The demo lets you edit the ratings by double clicking the cell._
+{{"demo": "pages/components/data-grid/editing/RenderRatingEditCellGrid.js", "bg": "inline", "defaultCodeOpen": false}}
+
+#### Edit using external button
```

Comment:
This is not in the TOC, it seems to be a missed opportunity for discoverability.

```suggestion
### Edit using external button
```